string id
bool online
float32 x
float32 y